### PR TITLE
Object bucket infinite pending regression

### PR DESCRIPTION
**Pull request by BlaineEXE**
**Description of your changes:**

Add an integration regression test for Ceph Object Storage Object Bucket
Claim (OBC) to verify that the OBC stays in "Bound" state after being
created. The claim should not fall back into "Pending" (or any other)
state soon after being created.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: tests/integration/ceph_base_object_test.go [hunk 1]
```diff
@@ -190,6 +194,27 @@ func runObjectE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite
 	require.Nil(s.T(), delobjErr)
 	logger.Infof("Objects deleted on bucket successfully")
 
+	// A bug exists in older versions of lib-bucket-provisioner that will revert a bucket and claim
+	// back to "Pending" phase after being created and initially "Bound" by looping infinitely in
+	// the bucket provision/creation loop. Verify that the OBC is "Bound" and stays that way. We
+	// need to verify this before the OBC is deleted, but we can avoid wasting extra time sleeping
+	// in a wait loop by completing all tests before this regression check.
+	logger.Infof("Regression check: Verify bucket does not revert to Pending phase")
```
Comment (leseb) [inline tests/integration/ceph_base_object_test.go:202]:
```suggestion
	logger.Info("Regression check: Verify bucket does not revert to Pending phase")
```

**Review by leseb — COMMENTED** [review]
nit, otherwise lgtm

File: tests/integration/ceph_base_object_test.go [hunk 2]
```diff
@@ -190,6 +194,27 @@ func runObjectE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite
...
+	timeToWaitForReversion := 2 * time.Minute
+	timeToCheckForReversion := timeBucketCreateVerified.Add(timeToWaitForReversion)
+	for time.Now().Before(timeToCheckForReversion) {
+		// simple loop to wait until the current time is after the time we should check for reversion
+		time.Sleep(5 * time.Second)
+	}
+	bktClient := bktClientset.New(k8sh.Clientset.RESTClient())
+	rest := bktClient.RESTClient()
+	obc := bktv1alpha1.ObjectBucketClaim{}
+	err = rest.Get().Resource("objectbucketclaim").Name(obcName).Do(ctx).Into(&obc)
```
Comment (thotz) [inline tests/integration/ceph_base_object_test.go:214]: Do OBC Status changes even before checking OB reference?

**Comment by leseb** [comment]
@BlaineEXE what's left on this one?

**Pull request closed without merging**